src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -120,10 +188,18 @@ public string GetDotNetFrameworkName(IFrameworkNameProvider mappings)
 
                 parts.Add(string.Format(CultureInfo.InvariantCulture, "Version=v{0}", GetDisplayVersion(framework.Version)));
 
-                if (!string.IsNullOrEmpty(framework.Profile))
+                if (!IsNet5Era && !string.IsNullOrEmpty(framework.Profile))
```

Comment:
Does this really matter?
Since we are not using profiles anymore, this should not be relevant.

Reply:
I'm just being doubleplus sure to express that Profiles are just not a thing in net5 in any way. Would you like me to remove this?

Reply:
Your call. 
We have not tried to do restrictions like this in the past because it complicates things. 
Granted right now we're being too permissive.

My main concern is readability though (given that this has no actual implementation impact because of how these properties are being set right now), so maybe a judgment call (or wait for someone else to complain, lol).